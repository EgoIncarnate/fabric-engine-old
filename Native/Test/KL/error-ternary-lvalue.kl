operator entry()
{
  var Integer a = 0, b = 0;
  (false? a: b) = 1;
  report a;
  report b;
}
