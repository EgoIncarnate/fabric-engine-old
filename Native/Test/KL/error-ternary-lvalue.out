GlobalList
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          AssignedVarDecl( VarDecl( 'a', Integer ), ConstSize( 0 ) )
            ConstSize( 0 )
          AssignedVarDecl( VarDecl( 'b', Integer ), ConstSize( 0 ) )
            ConstSize( 0 )
        ExprStatement
          AssignOp(ASSIGN_OP)
            TernaryOp(TERNARY_OP_COND)
              ConstBoolean( false )
              Var('a')
              Var('b')
            ConstSize( 1 )
        Report
          Var('a')
        Report
          Var('b')
4:4: error: assignment: the result of a ternary operation cannot be an l-value
